src/index.ts

```diff
@@ -221,11 +219,10 @@ export async function main(): Promise<void> {
           linkTemplateMap,
           frozenEnvironments,
         });
-        if (promotionsByTargetEnvironment) {
-          promotionsByFileThenEnvironment.set(
-            shortFilename(filename),
-            promotionsByTargetEnvironment,
-          );
+        if (environmentPromotions) {
+          promotionsByFile.set(shortFilename(filename), [
```

Comment:
I'm not sure I understand why the value in this map is a list. Does it ever have a non-1 length?